plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -1515,6 +1527,141 @@ var _ = Describe("validator", func() {
 				})
 			})
 
+			Context("admission plugin check", func() {
+				BeforeEach(func() {
+					shoot.Spec.Kubernetes.KubeAPIServer = &core.KubeAPIServerConfig{}
+				})
+
+				Context("KubeconfigSecretName is not present for admission plugin", func() {
+					It("should allow because KubeconfigSecretName is not present for admission plugin", func() {
+						shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins = []core.AdmissionPlugin{
+							{
+								Name: "plugin-1",
+							},
+						}
+
+						Expect(coreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&project)).To(Succeed())
+						Expect(coreInformerFactory.Core().InternalVersion().CloudProfiles().Informer().GetStore().Add(&cloudProfile)).To(Succeed())
+						Expect(coreInformerFactory.Core().InternalVersion().Seeds().Informer().GetStore().Add(&seed)).To(Succeed())
+						Expect(coreInformerFactory.Core().InternalVersion().SecretBindings().Informer().GetStore().Add(&secretBinding)).To(Succeed())
+
+						attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+						err := admissionHandler.Admit(ctx, attrs, nil)
+
+						Expect(err).To(BeNil())
+					})
+
+					It("should not allow because kubeconfig secret is not referenced in shoot .spec.resources", func() {
+						shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins = []core.AdmissionPlugin{
+							{
+								Name:                 "plugin-1",
+								KubeconfigSecretName: pointer.String("secret-1"),
+							},
+						}
+
+						Expect(coreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&project)).To(Succeed())
+						Expect(coreInformerFactory.Core().InternalVersion().CloudProfiles().Informer().GetStore().Add(&cloudProfile)).To(Succeed())
+						Expect(coreInformerFactory.Core().InternalVersion().Seeds().Informer().GetStore().Add(&seed)).To(Succeed())
+						Expect(coreInformerFactory.Core().InternalVersion().SecretBindings().Informer().GetStore().Add(&secretBinding)).To(Succeed())
+
+						attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+						err := admissionHandler.Admit(ctx, attrs, nil)
+
+						Expect(err).To(BeForbiddenError())
```

Comment:
Please check for the concrete error using `ContainSubstring` etc.
Similarly at other places.

Reply:
In this test suite, everywhere we are following the same thing. So for consistency I kept it like this

Reply:
I agree, but for new tests, we should assert the right error.

Reply:
If you want can open a separate PR for serializing all the tests in this suite.

Reply:
Sure.